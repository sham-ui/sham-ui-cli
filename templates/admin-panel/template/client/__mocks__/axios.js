const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            INTERCEPTORS.request( request );
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = {
    csrfToken: 'csrf-token',
    user: {
        Name: 'John Smith',
        Email: 'j.smith@example.com'
    }
};

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        this
            .use( 'get', '/csrftoken', '', 200, { 'x-csrf-token': defaultMocksData.csrfToken } )
            .use( 'get', '/validsession', defaultMocksData.user );
        return this;
    }
};
