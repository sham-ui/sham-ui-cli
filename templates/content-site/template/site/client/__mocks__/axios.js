const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            if ( INTERCEPTORS.request ) {
                INTERCEPTORS.request( request );
            }
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);


export default {
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data, params } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                if ( undefined !== params ) {
                    args.push( params );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        return this;
    }
};
