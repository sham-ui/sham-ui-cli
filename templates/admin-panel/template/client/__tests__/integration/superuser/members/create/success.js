import setup, { app } from '../../../helpers';
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
    setup();
} );

it( 'success create member', async() => {
    expect.assertions( 5 );

    axios
        .useDefaultMocks()
        .use( 'get', '/validsession', {
            ...axios.defaultMocksData.user,
            IsSuperuser: true
        } )
        .use( 'get', 'admin/members', {
            members: [],
            meta: {
                total: 0,
                offset: 0,
                limit: 50
            }
        } );

    history.pushState( {}, '', 'http://client.example.com/members' );
    await app.start();

    app.click( '[data-test-toggle-create-form]' );
    await app.waitRendering();

    app.checkMainPanel();

    const formData = {
        name: 'John Smith',
        email: 'john.smith@test.com',
        is_superuser: false,
        password: 'test'
    };

    axios
        .use( 'post', 'admin/members', {} )
        .use( 'get', 'admin/members', {
            members: [
                {
                    ID: 1,
                    Name: formData.name,
                    Email: formData.email,
                    IsSuperuser: formData.is_superuser
                }
            ],
            meta: {
                total: 1,
                offset: 0,
                limit: 50
            }
        } );

    app.form.fill( 'name', formData.name );
    app.form.fill( 'email', formData.email );
    app.form.fill( 'pass', formData.password );
    app.form.submit();

    app.click( '[data-test-modal] [data-test-ok-button]' );
    await app.waitRendering();
    app.checkMainPanel();

    expect( axios.mocks.post ).toHaveBeenCalledTimes( 1 );
    expect( axios.mocks.post.mock.calls[ 0 ][ 0 ] ).toBe( 'admin/members' );
    expect( axios.mocks.post.mock.calls[ 0 ][ 1 ] ).toEqual( formData );
} );

it( 'success create superuser member', async() => {
    expect.assertions( 5 );

    axios
        .useDefaultMocks()
        .use( 'get', '/validsession', {
            ...axios.defaultMocksData.user,
            IsSuperuser: true
        } )
        .use( 'get', 'admin/members', {
            members: [],
            meta: {
                total: 0,
                offset: 0,
                limit: 50
            }
        } );

    history.pushState( {}, '', 'http://client.example.com/members' );
    await app.start();

    app.click( '[data-test-toggle-create-form]' );
    await app.waitRendering();

    app.checkMainPanel();

    const formData = {
        name: 'John Smith',
        email: 'john.smith@test.com',
        is_superuser: true,
        password: 'test'
    };

    axios
        .use( 'post', 'admin/members', {} )
        .use( 'get', 'admin/members', {
            members: [
                {
                    ID: 1,
                    Name: formData.name,
                    Email: formData.email,
                    IsSuperuser: formData.is_superuser
                }
            ],
            meta: {
                total: 1,
                offset: 0,
                limit: 50
            }
        } );

    app.form.fill( 'name', formData.name );
    app.form.fill( 'email', formData.email );
    app.click( '[name="is_superuser"]' );
    app.form.fill( 'pass', formData.password );
    app.form.submit();

    app.click( '[data-test-modal] [data-test-ok-button]' );
    await app.waitRendering();
    app.checkMainPanel();

    expect( axios.mocks.post ).toHaveBeenCalledTimes( 1 );
    expect( axios.mocks.post.mock.calls[ 0 ][ 0 ] ).toBe( 'admin/members' );
    expect( axios.mocks.post.mock.calls[ 0 ][ 1 ] ).toEqual( formData );
} );
