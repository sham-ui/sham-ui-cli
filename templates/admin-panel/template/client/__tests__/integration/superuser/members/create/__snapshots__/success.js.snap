// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success create member 1`] = `
"<h4 class=\\"page-header\\"> Members
  <!--header-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"members-page\\">
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Hide create new member form</button>
      <div class=\\"panel mg-b-30 create-member\\">
        <form class=\\"form-layout\\">
          <!--header-->
          <!--if-->
          <!--ErrorsPanel-->
          <div class=\\"form-group\\">
            <label>Name:</label>
            <input name=\\"name\\" type=\\"text\\" autofocus=\\"\\">
          </div>
          <div class=\\"form-group\\">
            <label>Email:</label>
            <input name=\\"email\\" type=\\"text\\">
          </div>
          <div class=\\"form-group\\">
            <label></label>
            <label class=\\"checkbox\\">
              <input name=\\"is_superuser\\" type=\\"checkbox\\" value=\\"false\\"><span>Is Superuser</span></label>
          </div>
          <div class=\\"form-group\\">
            <label>Password:</label>
            <input name=\\"pass\\" type=\\"password\\">
          </div>
          <!--default-->
          <!--default-->
          <button type=\\"submit\\">
            <!--if-->Create
            <!--submit-->
          </button>
          <!--footer-->
        </form>
        <!--FormContainer-->
        <!--if-->
        <!--FormWithConfirmModal-->
      </div>
      <!--if-->
    </div>
    <div class=\\"container\\">
      <div class=\\"members-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
            <table>
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Name</th>
                  <th>Email</th>
                  <th>Is superuser?</th>
                  <th></th>
                </tr>
              </thead>
              <tbody></tbody>
            </table>
            <!--if-->
            <!--MembersTable-->
            <!--default-->
            <!--if-->
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
          </div>
        </div>
      </div>
      <!--if-->
    </div>
  </div>
  <!--body-->
</div>
<!--if-->
<!--Page-->
<!--ActivePageContainer-->"
`;

exports[`success create member 2`] = `
"<h4 class=\\"page-header\\"> Members
  <!--header-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"members-page\\">
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Show create new member form</button>
      <!--if-->
    </div>
    <div class=\\"container\\">
      <div class=\\"members-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
            <table>
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Name</th>
                  <th>Email</th>
                  <th>Is superuser?</th>
                  <th></th>
                </tr>
              </thead>
              <tbody>
                <tr class=\\"\\">
                  <td>1</td>
                  <td>John Smith</td>
                  <td>john.smith@test.com</td>
                  <td class=\\"text-center\\">No</td>
                  <td class=\\"actions\\">
                    <button data-test-update-button=\\"1\\"> Edit </button>
                    <button class=\\"btn-danger\\" data-test-delete-button=\\"1\\"> Delete </button>
                  </td>
                </tr>
              </tbody>
            </table>
            <!--if-->
            <!--MembersTable-->
            <!--default-->
            <!--if-->
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
          </div>
        </div>
      </div>
      <!--if-->
    </div>
  </div>
  <!--body-->
</div>
<!--if-->
<!--Page-->
<!--ActivePageContainer-->"
`;

exports[`success create superuser member 1`] = `
"<h4 class=\\"page-header\\"> Members
  <!--header-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"members-page\\">
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Hide create new member form</button>
      <div class=\\"panel mg-b-30 create-member\\">
        <form class=\\"form-layout\\">
          <!--header-->
          <!--if-->
          <!--ErrorsPanel-->
          <div class=\\"form-group\\">
            <label>Name:</label>
            <input name=\\"name\\" type=\\"text\\" autofocus=\\"\\">
          </div>
          <div class=\\"form-group\\">
            <label>Email:</label>
            <input name=\\"email\\" type=\\"text\\">
          </div>
          <div class=\\"form-group\\">
            <label></label>
            <label class=\\"checkbox\\">
              <input name=\\"is_superuser\\" type=\\"checkbox\\" value=\\"false\\"><span>Is Superuser</span></label>
          </div>
          <div class=\\"form-group\\">
            <label>Password:</label>
            <input name=\\"pass\\" type=\\"password\\">
          </div>
          <!--default-->
          <!--default-->
          <button type=\\"submit\\">
            <!--if-->Create
            <!--submit-->
          </button>
          <!--footer-->
        </form>
        <!--FormContainer-->
        <!--if-->
        <!--FormWithConfirmModal-->
      </div>
      <!--if-->
    </div>
    <div class=\\"container\\">
      <div class=\\"members-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
            <table>
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Name</th>
                  <th>Email</th>
                  <th>Is superuser?</th>
                  <th></th>
                </tr>
              </thead>
              <tbody></tbody>
            </table>
            <!--if-->
            <!--MembersTable-->
            <!--default-->
            <!--if-->
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
          </div>
        </div>
      </div>
      <!--if-->
    </div>
  </div>
  <!--body-->
</div>
<!--if-->
<!--Page-->
<!--ActivePageContainer-->"
`;

exports[`success create superuser member 2`] = `
"<h4 class=\\"page-header\\"> Members
  <!--header-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"members-page\\">
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Show create new member form</button>
      <!--if-->
    </div>
    <div class=\\"container\\">
      <div class=\\"members-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
            <table>
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Name</th>
                  <th>Email</th>
                  <th>Is superuser?</th>
                  <th></th>
                </tr>
              </thead>
              <tbody>
                <tr class=\\"\\">
                  <td>1</td>
                  <td>John Smith</td>
                  <td>john.smith@test.com</td>
                  <td class=\\"text-center\\">Yes</td>
                  <td class=\\"actions\\">
                    <button data-test-update-button=\\"1\\"> Edit </button>
                    <button class=\\"btn-danger\\" data-test-delete-button=\\"1\\"> Delete </button>
                  </td>
                </tr>
              </tbody>
            </table>
            <!--if-->
            <!--MembersTable-->
            <!--default-->
            <!--if-->
            <div class=\\"page-buttons\\">
              <!--if-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--if-->
            </div>
            <!--PaginationPages-->
          </div>
        </div>
      </div>
      <!--if-->
    </div>
  </div>
  <!--body-->
</div>
<!--if-->
<!--Page-->
<!--ActivePageContainer-->"
`;
